### PR TITLE
FIX: actually use ChainedDB

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
@scopatz ready for review!

File: regolith/chained_db.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def __getitem__(self, key):
         results = []
         # Try to get all the data from all the mappings
         for mapping in self.maps:
-            results.append(mapping.get(key, None))
+            results.append(mapping.get(key, "--bad-request--"))
```
Comment (scopatz) [inline regolith/chained_db.py]: I think you should define a different sentinal, something that can't possibly be a valid value,and preferably as singleton.

Check out xonsh's DefaultNotGiven for an example: https://github.com/xonsh/xonsh/blob/master/xonsh/tools.py#L232

File: regolith/chained_db.py [hunk 2]
```diff
@@ -27,8 +27,9 @@ def __getitem__(self, key):
             return list(itertools.chain(*results))
         else:
             for result in reversed(results):
-                if result is not None:
+                if result is not "--bad-request--":
```
Comment (scopatz) [inline regolith/chained_db.py]: Then this could be `result is not Singleton`

**Comment by scopatz** [comment]
Just the one bit about the singleton, but otherwise looks great!

**Comment by CJ-Wright** [comment]
@scopatz ready for review!

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**